src/ebay-fake-menu/menu-item.tsx

```diff
@@ -1,40 +1,81 @@
 import React, { ComponentProps, FC, useEffect, useRef } from 'react'
 import classNames from 'classnames'
 import { EbayIcon } from '../ebay-icon'
+import { EbayBadge } from '../ebay-badge'
 
-export type EbayFakeMenuItemProps = Omit<ComponentProps<'a'>, 'onKeyDown' | 'onMouseDown'> & {
-    current?: boolean;
-    disabled?: boolean;
-    autoFocus?: boolean;
-}
+export type EbayMenuItemType = 'button'
+export type EbayFakeMenuItemProps =
+    Omit<ComponentProps<'a'>, 'onKeyDown' | 'onMouseDown'> &
+    Omit<ComponentProps<'button'>, 'onKeyDown' | 'onMouseDown'> &
+    {
+        current?: boolean;
+        disabled?: boolean;
+        autoFocus?: boolean;
+        type?: EbayMenuItemType;
+        badgeNumber?: number;
+        badgeAriaLabel?: string;
+    }
 
 const EbayMenuItem: FC<EbayFakeMenuItemProps> = ({
     className,
     current,
     disabled,
     autoFocus,
+    type,
+    badgeNumber,
+    badgeAriaLabel,
     children,
     ...rest
 }) => {
-    const ref = useRef<HTMLAnchorElement>()
+    const ref = useRef<HTMLAnchorElement & HTMLButtonElement>()
+    const hasBadge = badgeNumber !== undefined
 
     useEffect(() => {
         if (autoFocus) {
             ref.current?.focus()
         }
     })
 
-    return (
+    const itemProps = {
+        ...rest,
+        ref,
+        className: classNames(
+            className,
+            'fake-menu__item',
+            hasBadge && 'menu__item--badged'),
+        'aria-label': badgeAriaLabel
+    }
+
+    const tick = <EbayIcon name="tick16" />
+    const badgeStyleFix = {
+        marginLeft: 'var(--spacing-100)',
+        marginRight: 'var(--spacing-100)'
+    }
+    const badge = hasBadge && <EbayBadge type="menu" number={badgeNumber} style={badgeStyleFix} />
```

Comment:
LGTM but maybe we can create another helper/util component called EbayBadgeTick to avoid overloading this component with edge cases Fix